Description: tweak the session system to make it for ubuntu.
The session contains:
 - ubuntu (unity), gnome-shell in the main gnome-session package.
 - gnome-fallback-compiz (compiz or other wm + gnome-panel), gnome-fallback (metacity +
   gnome-panel) in the gnome-session-fallback package.
 (TryExec enables to show them or not depends on the package installed)
Forwarded: Not needed

Index: gnome-session-3.4.1/data/gnome.desktop.in
===================================================================
--- gnome-session-3.4.1.orig/data/gnome.desktop.in	2010-02-09 08:22:01.000000000 -0500
+++ gnome-session-3.4.1/data/gnome.desktop.in	2012-05-06 20:22:30.285990734 -0400
@@ -1,7 +1,7 @@
 [Desktop Entry]
 _Name=GNOME
 _Comment=This session logs you into GNOME
-Exec=gnome-session
-TryExec=gnome-session
+Exec=gnome-session --session=gnome
+TryExec=gnome-shell
 Icon=
 Type=Application
Index: gnome-session-3.4.1/data/Makefile.am
===================================================================
--- gnome-session-3.4.1.orig/data/Makefile.am	2012-05-06 20:20:27.389381326 -0400
+++ gnome-session-3.4.1/data/Makefile.am	2012-05-06 20:20:29.177390192 -0400
@@ -9,7 +9,7 @@
 hwcompat_DATA = hardware-compatibility
 
 xsessiondir = $(datadir)/xsessions
-xsession_in_files = gnome.desktop.in gnome-fallback.desktop.in
+xsession_in_files = gnome.desktop.in gnome-fallback.desktop.in gnome-fallback-compiz.desktop.in ubuntu.desktop.in
 xsession_DATA = $(xsession_in_files:.desktop.in=.desktop)
 
 desktopdir = $(datadir)/applications
@@ -17,7 +17,7 @@
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 
 sessiondir = $(datadir)/gnome-session/sessions
-session_in_in_files = gnome.session.desktop.in.in gnome-fallback.session.desktop.in.in
+session_in_in_files = gnome.session.desktop.in.in gnome-fallback.session.desktop.in.in gnome-fallback-compiz.session.desktop.in.in ubuntu.session.desktop.in.in
 session_in_files = $(session_in_in_files:.session.desktop.in.in=.session.desktop.in)
 session_DATA = $(session_in_files:.session.desktop.in=.session)
 
Index: gnome-session-3.4.1/data/gnome-fallback-compiz.desktop.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ gnome-session-3.4.1/data/gnome-fallback-compiz.desktop.in	2012-05-06 20:20:29.177390192 -0400
@@ -0,0 +1,7 @@
+[Desktop Entry]
+_Name=GNOME Fallback
+_Comment=This session logs you into GNOME with the traditional panel
+Exec=gnome-session --session=gnome-fallback-compiz
+TryExec=gnome-session
+Icon=
+Type=Application
Index: gnome-session-3.4.1/data/gnome-fallback-compiz.session.desktop.in.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ gnome-session-3.4.1/data/gnome-fallback-compiz.session.desktop.in.in	2012-05-06 20:20:29.177390192 -0400
@@ -0,0 +1,8 @@
+[GNOME Session]
+_Name=GNOME Fallback
+RequiredComponents=gnome-panel;gnome-settings-daemon;
+RequiredProviders=windowmanager;
+DefaultProvider-windowmanager=gnome-wm
+DefaultProvider-notifications=notify-osd
+IsRunnableHelper=@LIBEXECDIR@/gnome-session-check-accelerated
+FallbackSession=gnome-fallback
Index: gnome-session-3.4.1/data/gnome-fallback.session.desktop.in.in
===================================================================
--- gnome-session-3.4.1.orig/data/gnome-fallback.session.desktop.in.in	2012-05-06 20:20:27.389381326 -0400
+++ gnome-session-3.4.1/data/gnome-fallback.session.desktop.in.in	2012-05-06 20:20:29.181390204 -0400
@@ -1,6 +1,6 @@
 [GNOME Session]
-_Name=GNOME fallback
+_Name=GNOME fallback (Safe Mode)
 RequiredComponents=gnome-panel;gnome-settings-daemon;
-RequiredProviders=windowmanager;notifications;
-DefaultProvider-windowmanager=gnome-wm
-DefaultProvider-notifications=notification-daemon
+RequiredProviders=windowmanager;
+DefaultProvider-windowmanager=metacity
+DefaultProvider-notifications=notify-osd
Index: gnome-session-3.4.1/data/gnome-fallback.desktop.in
===================================================================
--- gnome-session-3.4.1.orig/data/gnome-fallback.desktop.in	2012-05-06 20:20:27.389381326 -0400
+++ gnome-session-3.4.1/data/gnome-fallback.desktop.in	2012-05-06 20:20:29.181390204 -0400
@@ -1,7 +1,7 @@
 [Desktop Entry]
-_Name=GNOME Classic
-_Comment=This session logs you into GNOME
-Exec=gnome-session-fallback
+_Name=GNOME Fallback (No effects)
+_Comment=This session logs you into GNOME with the traditional panel without any graphical effect.
+Exec=gnome-session --session=gnome-fallback
 TryExec=gnome-session
 Icon=
 Type=Application
Index: gnome-session-3.4.1/data/ubuntu.session.desktop.in.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ gnome-session-3.4.1/data/ubuntu.session.desktop.in.in	2012-05-06 20:20:29.181390204 -0400
@@ -0,0 +1,6 @@
+[GNOME Session]
+_Name=Ubuntu
+RequiredComponents=gnome-settings-daemon;
+RequiredProviders=windowmanager;panel;
+DefaultProvider-windowmanager=compiz
+DefaultProvider-panel=compiz
Index: gnome-session-3.4.1/data/ubuntu.desktop.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ gnome-session-3.4.1/data/ubuntu.desktop.in	2012-05-06 20:20:29.181390204 -0400
@@ -0,0 +1,7 @@
+[Desktop Entry]
+_Name=Ubuntu
+_Comment=This session logs you into Ubuntu
+Exec=gnome-session --session=ubuntu
+TryExec=unity
+Icon=
+Type=Application
